github/security_advisories.go

```diff
@@ -47,6 +48,110 @@ type ListRepositorySecurityAdvisoriesOptions struct {
 	State string `url:"state,omitempty"`
 }
 
+// ListGlobalSecurityAdvisoriesOptions specifies the optional parameters to list the global security advisories.
+type ListGlobalSecurityAdvisoriesOptions struct {
+	ListCursorOptions
+
+	// If specified, only advisories with this GHSA (GitHub Security Advisory) identifier will be returned.
+	GHSAID string `url:"ghsa_id,omitempty"`
+
+	// If specified, only advisories of this type will be returned.
+	// By default, a request with no other parameters defined will only return reviewed advisories that are not malware.
+	// Default: reviewed
+	// Can be one of: reviewed, malware, unreviewed
+	Type string `url:"type,omitempty"`
+
+	// If specified, only advisories with this CVE (Common Vulnerabilities and Exposures) identifier will be returned.
+	CVEID string `url:"cve_id,omitempty"`
+
+	//If specified, only advisories for these ecosystems will be returned.
+	// Can be one of: actions, composer, erlang, go, maven, npm, nuget, other, pip, pub, rubygems, rust
+	Ecosystem string `url:"ecosystem,omitempty"`
```

Comment:
Every field with the `omitempty` tag should either be a pointer or a slice.  Please change them.  Line 77, for example, is fine as-it-is because it is a slice of strings.  A slice to structs, however, would need to be a slice to struct pointers.

Also, since I'm here, and this is a nit, please change `//If` to `// If`.

Reply:
thank you for your feedback!

Reply:
Sorry I wasn't clear before, as this request was not addressed in the last force-push.
I'll try to be more explicit this time.

Also, please do not use force-push, as it makes it more difficult for reviewers to see what changed since the last review.  We also always use squash-and-merge in this repo anyway, so you can have as many commits as you want and they will all be squashed down to a single commit when we merge.